src/main.c

```diff
@@ -453,12 +453,73 @@ static int dev_opt(const char *str) {
     !!strchr(str, '/');
 }
 
+static int cmp_ptr(const void *a, const void *b) {
+   return (*(int*)a - *(int*)b);
+}
+
+static int suggest_programmers(const char *programmer, LISTID programmers) {
+  const int max_distance = 3;
+
+  int idx = 0;
+  int pgmid_maxlen = 0;
+  typedef struct {
+    int distance;
+    char *pgmid;
+    char *desc;
+  } pgm_distance;
+
+  pgm_distance *d = malloc(1);
+  sort_programmers(programmers);
+  for(LNODEID ln1 = lfirst(programmers); ln1; ln1 = lnext(ln1)) {
+    PROGRAMMER *pgm = ldata(ln1);
+    if(!is_programmer(pgm))
+      continue;
+    for(LNODEID ln2 = lfirst(pgm->id); ln2; ln2 = lnext(ln2)) {
+      const char *id = ldata(ln2);
+      // Find the distance between the user specified programmer and id
+      int dist = levenshtein(programmer, id, 1, 1, 1, 1);
+      if(dist <= max_distance) {
+        pgm_distance *d_realloc = realloc(d, (idx+1)*sizeof(*d));
+        if(d_realloc)
+          d = d_realloc;
+        else {
+          pmsg_error("memory reallocation failed\n");
+          return -1;
+        }
+        d[idx].distance = dist;
+        d[idx].pgmid = cfg_strdup(__func__, id);
+        d[idx].desc = cfg_strdup(__func__, pgm->desc);
+        idx++;
+        if(strlen(id) > (size_t)pgmid_maxlen)
+          pgmid_maxlen = (int)strlen(id);
+      }
+    }
+  }
+  // Sort list so programmers with the smallest distance gets printed first
+  qsort(d, idx, sizeof(*d), cmp_ptr);
+  if(idx) {
```

Comment:
Could show the closest match only and all those that have the same distance as closest match.

Reply:
Thanks, done!